ml-agents/mlagents/trainers/sac/models.py

```diff
@@ -46,9 +46,41 @@ def __init__(
         self.activ_fn = self.swish
 
         self.policy_memory_in: Optional[tf.Tensor] = None
+        self.policy_memory_out: Optional[tf.Tensor] = None
```

Comment:
Not sure if there's a better way to group all of these.

Reply:
Ahh, this is ugly (one more reason to move to TF 2.0). I can take a stab at grouping by comment (e.g. divvy up policy vs. critic vs. output tensors) but they are all needed. 

I'm assuming we'll have a similar block in the PPO model?

Reply:
it's OK as it is. One way to DRY it up a bit would be something like
```
class QInfo
    self.q: Optional[tf.Tensor] = None
    self.q_p: Optional[tf.Tensor] = None
    self.q_memory_in: Optional[tf.Tensor] = None
    self.q_memory_out: Optional[tf.Tensor] = None
    self.q_heads: Optional[Dict[str, tf.Tensor]] = None
    self.q_pheads: Optional[Dict[str, tf.Tensor]] = None
    ....
class SACNetwork(LearningModel):
    self.q1 = QInfo()
    self.q2 = QInfo()
```
but that might not be a good abstraction (and probably not a good name).

Surprisingly it's not complaining about PPO right now.

Reply:
Hmm the class nesting doesn't seem very obvious. I'd almost want a group that is model inputs and one that is model outputs - but that seems to require more code changes, so let's leave it for now.